### PR TITLE
Fix svgCreate crashing (#2456)

### LINKED ISSUE #2456: svgCreate causes random crashes
**Describe the bug**

`svgCreate` crashes the game randomly when calling the function multiple times

**To reproduce**

/start svgtest
[svgtest.zip](https://github.com/multitheftauto/mtasa-blue/files/7727090/svgtest.zip)

**Screenshots**

Przechwytywanie

**Version**

Multi Theft Auto v1.5.9-release-21048

### PR DESCRIPTION
**Pull request by Lpsd**
Turns out you shouldn't create XML inside async task scheduler (specifically due to `CXMLArray::PopUniqueID` and stack ID stuff, that should only be done in main thread, ty sbx).

This PR generates the XML inside the Lua definition and passes it to the async task scheduler, instead of generating the XML inside the async task.

Resolves #2456

### CODE REVIEW

**Comment by Lpsd** [comment]
Any chance of a quick review? svgCreate using callback arg is kinda broken without this

**Comment by Inder00** [comment]
The fix works just fine for PoC with issue #2456, however I also encountered on the official build (before the fix) with another bug in the game shutdown situation.

image

**Comment by Lpsd** [comment]
Yeah, that crash is related - this PR fixes it.

**Review by Inder00 — APPROVED** [review]
LGTM

File: Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp [hunk 1]
```diff
@@ -56,6 +56,22 @@ bool CLuaVectorGraphicDefs::LoadFromData(lua_State* luaVM, CClientVectorGraphic*
...
+bool CLuaVectorGraphicDefs::LoadFromNode(lua_State* luaVM, CClientVectorGraphic* pVectorGraphic, CXMLNode* pNode)
+{
+    if (!luaVM || !pVectorGraphic)
+        return false;
+
+    if (!pVectorGraphic->SetDocument(pNode))
+    {
+        delete pVectorGraphic;
+
+        m_pScriptDebugging->LogCustom(luaVM, "Unable to load SVG data (check for XML syntax errors)");
```
Comment (Pirulax) [inline Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp:68]: image


No, like honestly, you're calling this from a secondary thread, this is a bomb waiting to explode.

**Comment by Pirulax** [comment]
I'm honestly surprised SVG works.
I just had a quick look at the source code, and it's pure luck.
There are so many calls to non thread-safe functions from secondary threads...
There will be more random crashes.
Good example is ` CLuaVectorGraphicDefs::SetSize` which is pushed as an async task in `CLuaVectorGraphicDefs::SVGSetSize`
Calling logging functions from secondary threads isn't good either.

**Pull request closed without merging**